plugins/modules/asset.py

```diff
@@ -90,9 +103,54 @@
 )
 
 
+def validate_module_params(params):
+    if params["state"] == "present":
+        if not params['builds']:
+            return "builds must include at least one element"
+    return None
+
+
+def _build_set(builds):
+    return {(
+        b.get('sha512'),
+        b.get('url'),
+        frozenset((b.get('headers', {}) or {}).items()),
+        frozenset(b.get('filters', []) or []),
+    ) for b in builds}
+
+
+def _do_builds_differ(current, desired):
+    if len(current) != len(desired):
+        return True
+
+    return _build_set(current) != _build_set(desired)
+
+
+def do_differ(current, desired):
+    if current is None:
+        return True
+
+    for key, value in desired.items():
+        current_value = current.get(key)
+        if key == 'builds':
+            if _do_builds_differ(current_value, value):
+                return True
+        elif value != current.get(key):
```

Comment:
Nitpick: a bit more efficient version would be `elif value != current_value:` since we already fetched it earlier.